src/dscanner/analysis/local_imports.d

```diff
@@ -5,99 +5,91 @@
 
 module dscanner.analysis.local_imports;
 
-import std.stdio;
-import dparse.ast;
-import dparse.lexer;
 import dscanner.analysis.base;
 import dscanner.analysis.helpers;
 import dsymbol.scope_;
 
+import std.stdio : writeln;
+
 /**
  * Checks for local imports that import all symbols.
  * See_also: $(LINK https://issues.dlang.org/show_bug.cgi?id=10378)
  */
-final class LocalImportCheck : BaseAnalyzer
+extern(C++) class LocalImportCheck(AST) : BaseAnalyzerDmd!AST
 {
-	alias visit = BaseAnalyzer.visit;
-
 	mixin AnalyzerInfo!"local_import_check";
+	alias visit = BaseAnalyzerDmd!AST.visit;
 
-	/**
-	 * Construct with the given file name.
-	 */
-	this(string fileName, const(Scope)* sc, bool skipTests = false)
-	{
-		super(fileName, sc, skipTests);
-	}
-
-	mixin visitThing!StructBody;
-	mixin visitThing!BlockStatement;
-
-	override void visit(const Declaration dec)
+	mixin ScopedVisit!(AST.FuncDeclaration);
+	mixin ScopedVisit!(AST.IfStatement);
+	mixin ScopedVisit!(AST.WhileStatement);
+	mixin ScopedVisit!(AST.ForStatement);
+	mixin ScopedVisit!(AST.ForeachStatement);
```

Comment:
What about struct/class declarations and template declarations? Aren't those supposed to be handled as well?

Reply:
Yes they should! Also I forgot to mention this here, but I think there is a pretty significant bug with this visitor in the actual D-Scanner: it won't throw the warning if it is inside a `unittest` block. I don't think that's the expected behavior. If I add the struct/class/template declarations, the tests (applying `D-Scanner` to phobos) will fail, and there is a significant amount of modifications I will need to make in phobos so the tests will pass. I can do them, just wanted to clarify if ignoring unittests every time is incorrect (which I think it is). What do you say? (Also the actual D-Scanner completly ignores templates for this check)

Reply:
Ok then, let's just stick with the current d-scanner implementation. Bug fixing should be done in later pass.

Reply:
Ok, if it looks good now, let me know, so I can squash the commits and we can merge this